examples/jenkins-image-sample.groovy

```diff
@@ -0,0 +1,119 @@
+    template.labels["mylabel"] = "myvalue"
+
+    // Process the modeled template. We could also pass JSON/YAML or a template name instead.
+    // note: -p option for oc process not in the oc version that we currently ship with openshift jenkins images
+    def objectModels = openshift.process( template )//, "-p", "MEMORY_LIMIT=600Mi")
+
+    // objectModels is a list of objects the template defined, modeled as Groovy objects
+    echo "The template will create ${objectModels.size()} objects"
+
+    // For fun, modify the objects that have been defined by processing the template
+    for ( o in objectModels ) {
+        o.metadata.labels[ "anotherlabel" ] = "anothervalue"
+    }
+
+    // Serialize the objects and pass them to the create API.
+    // We could also pass JSON/YAML directly; openshift.create(readFile('some.json'))
+    def created = openshift.create(objectModels)
+
+    // Create returns a selector which will always select the objects created
+    created.withEach {
+        // Each loop binds the variable 'it' to a selector which selects a single object
+        echo "Created ${it.name()} from template with labels ${it.object().metadata.labels}"
+    }
+
+    // Filter created objects and create a selector which selects only the new DeploymentConfigs
+    def dcs = created.narrow("dc")
+    echo "Database will run in deployment config: ${dcs.name()}"
+    timeout(5) {
+        // Find a least one pod related to the DeploymentConfig and wait it satisfies a condition
+        dcs.related('pods').untilEach(1) {
+            // untilEach only terminates when each selected item causes the body to return true
+            return it.object().status.phase != 'Pending'
+        }
+    }
+
+    // Print out all pods created by the DC
+    echo "Template created pods: ${dcs.related('pods').names()}"
+
+    // Show how we can use labels to select as well
+    echo "Finding dc using labels instead: ${openshift.selector('dc',[mylabel:'myvalue']).names()}"
+
+    echo "DeploymentConfig description"
+    dcs.describe()
+    echo "DeploymentConfig history"
+    dcs.rollout().history()
+
+    def rubySelector = openshift.selector("bc", "ruby")
+    def builds
+    try {
+        rubySelector.object()
+        builds = rubySelector.related( "builds" )
+    } catch (Throwable t) {
+        // The selector returned from newBuild will select all objects created by the operation
+        nb = openshift.newBuild( "https://github.com/openshift/ruby-hello-world", "--name=ruby" )
+
+        // Print out information about the objects created by newBuild
+        echo "newBuild created: ${nb.count()} objects : ${nb.names()}"
+        
+        // Filter non-BuildConfig objects and create selector which will find builds related to the BuildConfig
+        builds = nb.narrow("bc").related( "builds" )
+
+    }
+
+    // Raw watch which only terminates when the closure body returns true
+    builds.watch {
```

Comment:
well ... just noticed the timeout of 5 for the db deploy ... going to go with 7 for the overall job, unless you think different